open-banking-accelerator/components/consent-management/com.wso2.openbanking.accelerator.consent.extensions/src/test/java/com/wso2/openbanking/accelerator/consent/extensions/authorize/vrp/persistence/flow/ConsentPersistStepTests.java

```diff
@@ -0,0 +1,229 @@
+        doReturn(true).when(consentPersistDataMock).getApproval();
+
+        JSONObject payload = (JSONObject) parser
+                .parse(ConsentAuthorizeTestConstants.PAYLOAD_WITH_NON_STRING_ACCOUNTID);
+        doReturn(payload).when(consentPersistDataMock).getPayload();
+
+        consentPersistStep.execute(consentPersistDataMock);
+    }
+
+    @Test(priority = 9, expectedExceptions = ConsentException.class)
+    public void testCOFConsentPersistWithoutCOFAccount() throws Exception {
+
+        doReturn(consentDataMock).when(consentPersistDataMock).getConsentData();
+        doReturn(ConsentAuthorizeTestConstants.CONSENT_ID).when(consentDataMock).getConsentId();
+        doReturn(ConsentAuthorizeTestConstants.USER_ID).when(consentDataMock).getUserId();
+        doReturn(ConsentAuthorizeTestConstants.CLIENT_ID).when(consentDataMock).getClientId();
+        doReturn(consentResourceMock).when(consentDataMock).getConsentResource();
+        doReturn(ConsentAuthorizeTestConstants.getAuthResource()).when(consentDataMock).getAuthResource();
+        doReturn(ConsentAuthorizeTestConstants.FUNDS_CONFIRMATIONS).when(consentResourceMock)
+                .getConsentType();
+        doReturn(true).when(consentPersistDataMock).getApproval();
+
+        JSONObject payload = (JSONObject) parser
+                .parse(ConsentAuthorizeTestConstants.COF_PERSIST_PAYLOAD_WITHOUT_COF_ACC);
+        doReturn(payload).when(consentPersistDataMock).getPayload();
+
+        consentPersistStep.execute(consentPersistDataMock);
+    }
+
+    @Test(priority = 10, expectedExceptions = ConsentException.class)
+    public void testCOFConsentPersistWithNonStringCOFAccount() throws Exception {
+
+        doReturn(consentDataMock).when(consentPersistDataMock).getConsentData();
+        doReturn(ConsentAuthorizeTestConstants.CONSENT_ID).when(consentDataMock).getConsentId();
+        doReturn(ConsentAuthorizeTestConstants.USER_ID).when(consentDataMock).getUserId();
+        doReturn(ConsentAuthorizeTestConstants.CLIENT_ID).when(consentDataMock).getClientId();
+        doReturn(consentResourceMock).when(consentDataMock).getConsentResource();
+        doReturn(ConsentAuthorizeTestConstants.getAuthResource()).when(consentDataMock).getAuthResource();
+        doReturn(ConsentAuthorizeTestConstants.FUNDS_CONFIRMATIONS).when(consentResourceMock)
+                .getConsentType();
+        doReturn(true).when(consentPersistDataMock).getApproval();
+
+        JSONObject payload = (JSONObject) parser
+                .parse(ConsentAuthorizeTestConstants.COF_PERSIST_PAYLOAD_WITH_NON_STRING_COF_ACC);
+        doReturn(payload).when(consentPersistDataMock).getPayload();
+
+        consentPersistStep.execute(consentPersistDataMock);
+    }
+
+    @Test(priority = 11, expectedExceptions = ConsentException.class)
+    public void testCOFPersistThrowingExceptionWhenConsentBinding() throws Exception {
```

Comment:
Shall we add a test with happy path too?, (Test without expecting an exception)

Reply:
Fixed in [f1f5354]